mobius-android/src/main/java/com/spotify/mobius/android/MobiusAndroidViewModel.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * -\-\-
+ * Mobius
+ * --
+ * Copyright (c) 2017-2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.mobius.android;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+import com.spotify.mobius.First;
+import com.spotify.mobius.Init;
+import com.spotify.mobius.MobiusLoop;
+import com.spotify.mobius.MobiusLoop.Factory;
+import com.spotify.mobius.functions.Function;
+import javax.annotation.Nonnull;
+
+/**
+ * A Mobius Loop controller which is based on the Android ViewModel. <br>
+ *
+ * <p>This controller has the concept of a View Effect (parameter V) which is a type of effect that
+ * requires the corresponding Android lifecycle owner to be in an active state i.e. between onResume
+ * and onPause. To allow the normal effect handler to send these, the controller will provide a
+ * Consumer of these View Effects to the Loop Factory Provider, which can then be passed into the
+ * normal Effect handler so it can delegate view effects where necessary<br>
+ *
+ * <p>Since it's based on Android View model, this controller will keep the loop alive as long as
+ * the lifecycle owner it is associated with (via a factory to produce it) is not destroyed -
+ * meaning the Mobius loop will persist through rotations and brief app minimization to background,
+ * <br>
+ *
+ * <p>While the loop is running but the view is paused, which is between onPause and onDestroy, the
+ * controller will keep the latest model/state sent by the loop and will keep a queue of View
+ * Effects that have been sent by the effect handler. The loop is automatically disposed when the
+ * lifecycle owner is destroyed
+ *
+ * @param <M> The Model with which the Mobius Loop will run
+ * @param <E> The Event type accepted by the loop
+ * @param <F> The Effect type handled by the loop
+ * @param <S> The View State which will be emitted by this controller
+ * @param <V> The View Effect which will be emitted by this controller
+ */
+public class MobiusAndroidViewModel<M, E, F, S, V> extends ViewModel {
```

Comment:
I thought some more about this name - maybe it'd be a good idea to call it MobiusLoopViewModel after all, to avoid confusing it with the AndroidViewModel class from AAC (which has nothing to do with this class)

Reply:
👍 - will rename to that, seems good